src/main/java/com/hazelcast/aws/AwsRequestUtils.java

```diff
@@ -83,7 +83,9 @@ private static void addComponents(List<String> components, Map<String, String> a
     private static String urlEncode(String string) {
         String encoded;
         try {
-            encoded = URLEncoder.encode(string, "UTF-8").replace("+", "%20");
+            encoded = URLEncoder.encode(string, "UTF-8")
+                    .replace("+", "%20")
+                    .replace("*", "%2A");
```

Comment:
I wonder, why this change is needed?

Reply:
In the [documentation for signing AWS requests](https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html), they write that only unreserved characters defined by RFC 3986 should not be URI-encoded. `*` is not part of these characters so it should be "Percent-encode" as another part of the documentation specifies.

Reply:
thanks for the explanation 👍